src/Moq/StringBuilderExtensions.cs

```diff
@@ -41,18 +41,31 @@ public static StringBuilder AppendNameOf(this StringBuilder stringBuilder, Metho
 
 			if (includeGenericArgumentList && method.IsGenericMethod)
 			{
-				stringBuilder.Append('<');
-				var genericArguments = method.GetGenericArguments();
-				for (int i = 0, n = genericArguments.Length; i < n; ++i)
-				{
-					if (i > 0)
-					{
-						stringBuilder.Append(", ");
-					}
-					stringBuilder.AppendNameOf(genericArguments[i]);
-				}
+				stringBuilder.AppendGenericArguments(method);
+			}
+
+			return stringBuilder;
+		}
+
+		public static StringBuilder AppendNameOfAddEvent(this StringBuilder stringBuilder, MethodBase method, bool includeGenericArgumentList)
+		{
+			stringBuilder.Append(method.Name.Replace("add_", "")).Append(" += ");
```

Comment:
This line could be merged to `stringBuilder.Append(method.Name.Replace("add_", " += "));`